pkg/apis/agones/v1/gameserver.go

```diff
@@ -456,6 +457,24 @@ func (gss *GameServerSpec) Validate(devAddress string) ([]metav1.StatusCause, bo
 	return causes, len(causes) == 0
 }
 
+// ValidateResource validates limit or Memory CPU resources used for containers in pods
+// If a GameServer is invalid there will be > 0 values in
+// the returned array
+func ValidateResource(request resource.Quantity, limit resource.Quantity, resourceName corev1.ResourceName) []error {
```

Comment:
I have placed ValidateResource() here so we will check PodTemplate in a Fleet and GameServer using exact same function.